### PR TITLE
(IMAGES-721) Restructure Windows Template Dirs

### PR DESCRIPTION
**Pull request by jcoconnor**
This remodels the Windows Templates to follow the new Linux practice of
sub-grouping all the platform version variants under each major OS
family.

The puppet and windows scripts files have also been moved under the
templates/windows directories as these are specific to the windows
builds only.

### CODE REVIEW

File: templates/windows/win/10-ent/i386/vars.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
     "template_name"     : "win-10-ent-i386",
+    "packer_proc_arch"  : "x86",
```
[the remark this answers is not in the document]
Comment (jcoconnor) [reply templates/windows/win/10-ent/i386/vars.json]: Thats for the autounattend.xml file - I maybe should name the variable better.
Arch is either ```amd64``` or ```x86```

**Comment by ekinanp** [comment]
Already talked about this on HipChat, but re-posting here:

```
[8:05 AM] Enis Inan: ok. one small nitpick is that the "files" directory in each osverarch directory should be under a "common" directory so it is "common/files" instead of "files" -- also, that directory should be in "os-group/os-ver" instead of osverarch. the thinking is that it's likely those files are shared across multiple architectures
[8:06 AM] Enis Inan: like in win-10ent, the platform-packages script is the same script for i386 and x86_64
```

**Comment by jcoconnor** [comment]
@ekinanp The files in the ```osverarch``` directory are actually intended to be specific to the architecture - potentially you could have different patches to apply depending on the arch, so going to leave that as is.

**Comment by jcoconnor** [comment]
This is ready for review/merge.
cc @ekinanp @McdonaldSeanp @puppetlabs/windows

**Pull request merged**